tensorboard/webapp/widgets/custom_modal/custom_modal_test.ts

```diff
@@ -135,4 +136,59 @@ describe('custom modal', () => {
       expect(fixture.componentInstance.isOpen).toBeFalse();
     });
   });
+
+  describe('ensures content is always within the window', () => {
+    beforeEach(() => {
+      window.innerHeight = 1000;
+      window.innerWidth = 1000;
+    });
+
+    it('sets left to 0 if less than 0', async () => {
+      const fixture = TestBed.createComponent(TestableComponent);
+      fixture.detectChanges();
+      fixture.componentInstance.openAtPosition({x: -10, y: 0});
+      expect(fixture.componentInstance.isOpen).toBeFalse();
+      await waitFrame();
+      fixture.detectChanges();
+
+      const content = fixture.debugElement.query(By.css('.content'));
+      expect(content.nativeElement.style.left).toEqual('0px');
+    });
+
+    it('sets top to 0 if less than 0', async () => {
+      const fixture = TestBed.createComponent(TestableComponent);
+      fixture.detectChanges();
+      fixture.componentInstance.openAtPosition({x: 0, y: -10});
+      expect(fixture.componentInstance.isOpen).toBeFalse();
+      await waitFrame();
+      fixture.detectChanges();
+
+      const content = fixture.debugElement.query(By.css('.content'));
+      expect(content.nativeElement.style.top).toEqual('0px');
+    });
+
+    it('sets left to maximum if content overflows the window', async () => {
+      const fixture = TestBed.createComponent(TestableComponent);
+      fixture.detectChanges();
+      fixture.componentInstance.openAtPosition({x: 1010, y: 0});
+      expect(fixture.componentInstance.isOpen).toBeFalse();
+      await waitFrame();
+      fixture.detectChanges();
+      const content = fixture.debugElement.query(By.css('.content'));
+      // While rendering in a test the elements width and height will appear to be 0.
```

Comment:
Well that is annoying. I would really like to ensure this is window width - elements width. If it is too difficult this will do.

Reply:
I totally agree. Unfortunately even hard coding styles into this is resulting in width and height of 0
![image](https://github.com/tensorflow/tensorboard/assets/78179109/ddf0a648-6256-4597-9053-3306a42ae09a)

![image](https://github.com/tensorflow/tensorboard/assets/78179109/7902482f-2a91-4683-b9d0-d44817586d9c)